### PR TITLE
Add flag to remove plugins

### LINKED ISSUE #88: Tool deleted all my plugins
### What happened?

I had a bunch of plugins installed. The tool ran `pulumi plugin rm --all`

This wasn't ideal. It even deleted my language plugins.

### Expected Behavior

It doesn't delete all my plugins

### Steps to reproduce

Run the tool

### Output of `pulumi about`

N/A

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by aq17**
Fixes #88

### CODE REVIEW

**Comment by aq17** [comment]
Going off of https://pulumi.slack.com/archives/C051T068ASW/p1684867158092519, we can make removing plugins optional and false by default, and then set `remove-plugins: true` in all of our provider config files. 

Suggestions on a more helpful flag description/ warning if plugins are not removed would be great 😃

File: main.go [hunk 1]
```diff
@@ -168,9 +168,14 @@ If the passed version does not exist, an error is signaled.`)
...
 
 	cmd.PersistentFlags().StringVar(&context.UpstreamProviderName, "upstream-provider-name", "",
-		`The name of the upstream provider. 
+		`The name of the upstream provider.
 Required unless running from provider root and set in upgrade-config.yml.`)
 
+	cmd.PersistentFlags().BoolVar(&context.RemovePlugins, "remove-plugins", false,
+		`Remove all pulumi plugins from cache before running the upgrade.
+		It is possible that the generated examples may be non-deterministic depending on which
+		plugins are used if existing versions are present in the cache.`)
```
Comment (guineveresaenger) [inline main.go:177]: this is a good error message, tbqh! very informative

File: main.go [hunk 2]
```diff
@@ -168,9 +168,14 @@ If the passed version does not exist, an error is signaled.`)
 - "autoalias": Apply auto aliasing to the provider.`)
 
 	cmd.PersistentFlags().StringVar(&context.UpstreamProviderName, "upstream-provider-name", "",
-		`The name of the upstream provider. 
+		`The name of the upstream provider.
 Required unless running from provider root and set in upgrade-config.yml.`)
 
+	cmd.PersistentFlags().BoolVar(&context.RemovePlugins, "remove-plugins", false,
```
Comment (guineveresaenger) [inline main.go:174]: this means a corresponding update in the upgrade-provider-action, and ci-mgmt, yes?

**Review by guineveresaenger — APPROVED** [review]
LGTM!

File: main.go (hunk 2 above)
Comment (aq17) [reply main.go:174]: Yup, add `remove-plugins: true` to ci-mgmt for our providers' config file, and an input to the action

**Pull request merged**